### PR TITLE
Correct MaxPool documentation

### PR DESCRIPTION
**Pull request by jcwchen**
**Description**
The formula for output shape of MaxPool was wrong...  Should be as follows: `output_spatial_shape[i] = floor((input_spatial_shape[i] + pad_shape[i] - {kernelSpatialShape}) / strides_spatial_shape[i] - 1)`

Reference:
PyTorch: https://pytorch.org/docs/stable/generated/torch.nn.MaxPool2d.html
ONNXRuntime: https://github.com/microsoft/onnxruntime/blob/c99aa3a3f3834adcbb888ce4b964f2695c524eae/onnxruntime/core/providers/cpu/nn/pool_attributes.h#L178

**Motivation and Context**
https://github.com/onnx/onnx/issues/3523

### CODE REVIEW

File: onnx/defs/nn/defs.cc [hunk 1]
```diff
@@ -227,11 +227,11 @@ std::function<void(OpSchema&)> PoolOpSchemaGenerator(
  subset of the input tensor according to the kernel size and downsampling the
  data into the output tensor Y for further processing. The output spatial shape will be following:
  ```
- output_spatial_shape[i] = floor((input_spatial_shape[i] + pad_shape[i] - {kernelSpatialShape}) / strides_spatial_shape[i] + 1)
+ output_spatial_shape[i] = floor((input_spatial_shape[i] + pad_shape[i] - {kernelSpatialShape}) / strides_spatial_shape[i] - 1)
```
Comment (gramalingam) [inline onnx/defs/nn/defs.cc:230]: Doesn't sound right. Consider the simple case where there is no padding, stride is 1, and kernel-size is same as input-size. We expect the output size to be 1, right? So, the original "+1" sounds right, not "-1". Can you double-check? Thanks!

Comment (gramalingam) [reply onnx/defs/nn/defs.cc:230]: I responded to the original issue, please take a look.

**Comment by jcwchen** [comment]
Close it now since it is not correct.
